src/main/src/main/java/org/geoserver/config/GeoServerLoaderListener.java

```diff
@@ -0,0 +1,35 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.config;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.config.util.XStreamPersister;
+
+/**
+ * Listener interface for {@linkplain DefaultGeoServerLoader} load catalog and geoServer startup
+ * process.
+ */
+public interface GeoServerLoaderListener {
+
+    public static final GeoServerLoaderListener EMPTY_LISTENER = new EmptyGeoServerLoaderListener();
+
+    void loadCatalog(Catalog catalog, XStreamPersister xp);
+
+    void loadGeoServer(GeoServer geoServer, XStreamPersister xp);
+
+    default boolean isEmpty() {
```

Comment:
Not sure this is useful. The empty listener does nothing anyways? The check has to call this, which actually does _some_ work, making things slower than just calling the empty method.

Reply:
Makes sense, removed the isEmpty method and using the empty instance methods instead checking this method now.